### ISSUE TITLE
Support filename stored in $FIL element

### ISSUE THREAD

**Issue by MSHelm** [saeyslab/PeacoQC#2]
Hey everyone,
since we already met at CIMT and CYTO this year I am currently testing your package and absolutely love it.
I realized that you use the `FILENAME` element in the description slot in several function to determine the filename. Would it be possible to also support the `$FIL` element, which is the FCS format equivalent for storing the filename? You could also check for both using something like below (notice that I use `nchar` instead of` length > 0`, as `length(NA) = 1`)

```
  if (nchar(flowCore::keyword(ff)[["$FIL"]]) > 0) {
      filename <- basename(flowCore::keyword(ff)[["$FIL"]])
  } else if (nchar(flowCore::keyword(ff)$FILENAME) > 0) {
      filename <- basename(flowCore::keyword(ff)$FILENAME)
  }
```

Thanks a ton and can't wait to see your paper!
Best
Martin

**Comment by AnneliesEmmaneel** [comment]
Hi Martin,

Thank you for your suggestion and your interest in the package. I now changed every function in that way that it will check in both description slots if there is a name available. (Version 1.3.2)

If you have more suggestions or questions, you can always let me know.

With kind regards,

Annelies

**Comment by MSHelm** [comment]
Hi @AnneliesEmmaneel 
thanks for incorporating my suggestion :) I tried it and saw that we both made an error.
`nchar(NA)` does not return 0, as I thought, but instead NA, which causes an error in the if clause.
And I think you switched up the contents of the if clause, so if the content of `FILENAME` is there it then uses the content of `$FIL`

The following code should be correct:
```
if ((length(flowCore::keyword(ff)$FILENAME) > 0) && !is.na(flowCore::keyword(ff)$FILENAME)) {
        filename <- basename(flowCore::keyword(ff)$FILENAME)
    }
    else if ((length(flowCore::keyword(ff)[["$FIL"]]) > 0) &&  !is.na(flowCore::keyword(ff)[["$FIL"]])) {
        filename <- basename(flowCore::keyword(ff)[["$FIL"]])
    }
```

Sorry for the confusion I caused :(

**Comment by AnneliesEmmaneel** [comment]
Hi,

I tried to fix it and it still works in my version (1.3.3). Could you check this as well?
Thank you in advance!

Kind regards,
Annelies

**Comment by MSHelm** [comment]
Hi Annelies,
sorry for the late reply, everything works fine now. Thanks so much and congratulations on the paper!
Best
Martin
